src/qlik-date-picker.js

```diff
@@ -48,21 +48,33 @@ define(["qlik", "jquery", "./lib/moment.min", "./calendar-settings", "css!./lib/
             });
             return dateStates;
         }
+        function isEmpty(obj) {
+            for(var key in obj) {
+                if(obj.hasOwnProperty(key))
+                    return false;
+            }
+            return true;
+        }
         function createHtml(dateStates, DateFormat, props) {
             var html = '<div>'
```

Comment:
When the date picker does not have a Date Field this `<div>` tag will not be closed by corresponding `</div>`. Suggest to move line 73 to the bottom, outside of if-else statement.

Reply:
done !